editor/project_manager.cpp

```diff
@@ -281,6 +284,19 @@ void ProjectDialog::_path_text_changed(const String &p_path) {
 	}
 }
 
+void ProjectDialog::_path_text_changed(const String &p_path) {
+	Ref<DirAccess> d = DirAccess::create(DirAccess::ACCESS_FILESYSTEM);
+	String p = p_path;
+	if (mode == MODE_IMPORT && ((p.ends_with(".zip") && d->file_exists(p)) || (p.strip_edges().ends_with(".zip") && d->file_exists(p.strip_edges())))) {
```

Comment:
I'd suggest making a helper to simplify this code:

```
static bool is_zip_file(Ref<DirAccess>& d, String& path)
{
	return path.ends_with(".zip") && d->file_exists(path);
}
```

Then this check becomes more readable:
```
	if (mode == MODE_IMPORT
		&& (is_zip_file(d, p) || (is_zip_file(d, p.strip_edges()))))
	{
```

And the above `is_zip` init becomes simpler too.

Reply:
Thank you for your suggestion

Reply:
You're right that it's not quite equivalent to the init code, but that code is wrong. Writing it like this would correctly handle leading spaces in the path:

```
	} else if (is_zip_file(d, project_path->get_text())) {
		valid_path = project_path->get_text();
		is_zip = true;
	} else if (is_zip_file(d, project_path->get_text().strip_edges())) {
		valid_path = project_path->get_text().strip_edges();
		is_zip = true;
	}
```

In the existing code, the case `else if (project_path->get_text().ends_with(".zip"))` will incorrectly entered for `  c:/therewerespaces.zip` which might be valid but has incorrect leading spaces. That would prevent it from entering the next case `else if (project_path->get_text().strip_edges().ends_with(".zip"))` which handles stripping non ASCII characters.